File: api/bot.py
import json
import urllib.request

# Telegram bot token
TOKEN = "YOUR_BOT_TOKEN"
TELEGRAM_API = f"https://api.telegram.org/bot{TOKEN}/sendMessage"

# Telegra.ph API
TELEGRAPH_API = "https://api.telegra.ph/createPage"

def send_telegram(chat_id, text):
    data = {
        "chat_id": chat_id,
        "text": text
    }
    req = urllib.request.Request(TELEGRAM_API, data=json.dumps(data).encode(), headers={'Content-Type': 'application/json'})
    urllib.request.urlopen(req)

def publish_telegra(text, author_name="Bot"):
    lines = text.strip().split("\n")
    title = lines[0]
    content_lines = lines[1:]
    content = [{"tag": "p", "children": [line]} for line in content_lines]
    payload = {
        "title": title,
        "author_name": author_name,
        "content": content,
        "return_content": False
    }
    req = urllib.request.Request(TELEGRAPH_API, data=json.dumps(payload).encode(), headers={'Content-Type': 'application/json'})
    resp = urllib.request.urlopen(req)
    result = json.loads(resp.read())
    if result.get("ok"):
        return result["result"]["url"]
    else:
        return None

def handler(event, context):
    body = json.loads(event['body'])
    message = body.get("message", {})
    text = message.get("text", "")
    chat_id = message["chat"]["id"]
    if not text:
        send_telegram(chat_id, "请发送文本")
        return {"statusCode": 200, "body": "ok"}

    url = publish_telegra(text, author_name=message["from"]["first_name"])
    if url:
        send_telegram(chat_id, f"文章已发布: {url}")
    else:
        send_telegram(chat_id, "发布失败")
    return {"statusCode": 200, "body": "ok"}

File: api/bot.js
import https from "https";

// 直接写死你的 Bot Token
const TELEGRAM_TOKEN = "6055255911:AAHBLN50Aye0j7Lrmn-EqnsIuf4dVDUkPz4";

export default async function handler(req, res) {
  if (req.method !== "POST") {
    return res.status(200).send("Bot running");
  }

  let body;
  try {
    body = typeof req.body === "object" ? req.body : JSON.parse(req.body);
  } catch (e) {
    return res.status(400).send("Invalid JSON");
  }

  // 只处理文本消息
  if (body.message && body.message.text) {
    const chatId = body.message.chat.id;
    const text = body.message.text;

    // 发送确认消息
    sendMessage(chatId, "✅ 我收到了你的消息: " + text);
  }

  res.status(200).json({ ok: true });
}

// 发送消息函数
function sendMessage(chatId, text) {
  const data = JSON.stringify({ chat_id: chatId, text });
  const options = {
    hostname: "api.telegram.org",
    path: `/bot${TELEGRAM_TOKEN}/sendMessage`,
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "Content-Length": data.length
    }
  };

  const req = https.request(options);
  req.write(data);
  req.end();
}
